mungegithub/mungers/retest-not-required.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"regexp"
+
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+// RetestMunger will label a PR with retest-not-required if it only affects
+// .md files.
+// This may eventually be rewritten using a better heuristic for deciding
+// which filetypes require retesting.
+type RetestMunger struct{}
+
+const (
+	retestCommentBody = "Retest not required for this PR because of changed filetypes."
+)
+
+var (
+	markdownRE = regexp.MustCompile(`.+\.md$`)
+)
+
+func init() {
+	r := &RetestMunger{}
+	RegisterMungerOrDie(r)
+	RegisterStaleComments(r)
+}
+
+// Initialize will initialize the munger
+func (r *RetestMunger) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// Name is the name usable in --pr-mungers
+func (r *RetestMunger) Name() string { return retestNotRequiredLabel }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (r *RetestMunger) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// EachLoop is called at the start of every munge loop
+func (r *RetestMunger) EachLoop() error { return nil }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (r *RetestMunger) RequiredFeatures() []string { return []string{} }
+
+// Munge is the workhorse the will actually make updates to the PR
+func (r *RetestMunger) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
```

Comment:
maybe check if the label is already applied first?

Also, I think there's one more step: if the label has been applied AND it was applied by the mergebot AND the PR now has a non-md file change in it, then we should probably remove the label. (just don't want to remove it if a human put it on manually)